packages/near-fast-auth-signer/src/utils/multi-chain/chains/Bitcoin.ts

```diff
@@ -0,0 +1,448 @@
+    const s = signature.s.padStart(64, '0');
+
+    const rawSignature = Buffer.from(r + s, 'hex');
+
+    if (rawSignature.length !== 64) {
+      throw new Error('Invalid signature length.');
+    }
+
+    return rawSignature;
+  }
+
+  /**
+   * Sends a signed transaction to the Bitcoin network.
+   * This function takes the hexadecimal representation of a signed transaction
+   * and broadcasts it to the network using a proxy URL to bypass CORS restrictions.
+   * It logs the transaction ID if the broadcast is successful, or an error message otherwise.
+   *
+   * @param {string} txHex - The hexadecimal string of the signed transaction.
+   * @param {Object} [options] - Optional parameters.
+   * @param {boolean} [options.proxy=false] - Whether to use a proxy URL for the transaction broadcast.
+   * @returns {Promise<string>} A promise that resolves with the txid once the transaction is successfully broadcast
+   */
+  async sendTransaction(
+    txHex: string,
+    options?: { proxy?: boolean }
+  ): Promise<string | undefined> {
+    try {
+      const proxyUrl = options?.proxy ? 'https://corsproxy.io/?' : '';
+      const response = await axios.post(
+        `${proxyUrl}${this.providerUrl}tx`,
+        txHex
+      );
+
+      if (response.status === 200) {
+        return response.data;
+      }
+      throw new Error(`Failed to broadcast transaction: ${response.data}`);
+    } catch (error) {
+      throw new Error(`Error broadcasting transaction: ${error}`);
+    }
+  }
+
+  /**
+   * Calculates the fee properties for a Bitcoin transaction.
+   * This function fetches the Unspent Transaction Outputs (UTXOs) for the given address,
+   * and the fee rate for the specified confirmation target. It then uses the `coinselect` algorithm
+   * to select the UTXOs to be spent and calculates the fee required for the transaction.
+   *
+   * @param {string} from - The Bitcoin address from which the transaction is to be sent.
+   * @param {Array<{address: string, value: number}>} targets - An array of target addresses and values (in satoshis) to send.
+   * @param {number} [confirmationTarget=6] - The desired number of blocks in which the transaction should be confirmed.
+   * @returns {Promise<{inputs: UTXO[], outputs: {address: string, value: number}[], fee: number}>} A promise that resolves to an object containing the inputs (selected UTXOs), outputs (destination addresses and values), and the transaction fee in satoshis.
+   */
+  async getFeeProperties(
+    from: string,
+    targets: {
+      address: string;
+      value: number;
+    }[],
+    confirmationTarget = 6
+  ): Promise<{
+    inputs: UTXO[],
+    outputs: {address: string, value: number}[],
+    fee: number,
+  }> {
+    const utxos = await this.fetchUTXOs(from);
+    const feeRate = await this.fetchFeeRate(confirmationTarget);
+
+    // TODO: test net won't work with feeRate fetched, it need to add 1
+    const ret = coinselect(utxos, targets, feeRate + 1);
```

Comment:
if + 1 is only todo with testnet, can we implement a logic to check if `config.networkType === 'testnet' ? feeRate + 1 : feeRate`

Reply:
Good point... I'm not sure if mainnet has the same problem. I only tested on testnet as mainnet need real funds. 

I will update the comment